platform/posix/transport/src/openssl_posix.c

```diff
@@ -637,16 +644,27 @@ int32_t Openssl_Send( NetworkContext_t * pNetworkContext,
                       size_t bytesToSend )
 {
     int32_t bytesSent = 0;
+    int32_t sslError = 0;
 
-    /* SSL write of data. */
-    bytesSent = ( int32_t ) SSL_write( pNetworkContext->pSsl,
-                                       pBuffer,
-                                       bytesToSend );
+    if( pNetworkContext->pSsl != NULL )
+    {
+        /* SSL write of data. */
+        bytesSent = ( int32_t ) SSL_write( pNetworkContext->pSsl,
+                                           pBuffer,
+                                           bytesToSend );
 
-    if( bytesSent <= 0 )
+        if( bytesSent <= 0 )
+        {
+            sslError = SSL_get_error( pNetworkContext->pSsl, bytesSent );
+
+            LogError( ( "Failed to send data over network: SSL_write of OpenSSL failed: "
+                        " ErrorStatus=%s.", ERR_reason_error_string( sslError ) ) );
+        }
+    }
+    else
     {
-        LogError( ( "SSL_write of OpenSSL failed to send data: "
-                    " status=%d.", bytesSent ) );
+        LogError( ( "Failed to send data over network: "
+                    "Network context has invalid SSL object." ) );
```

Comment:
Would it be better to say NULL instead of invalid?

Reply:
Updated